lms/services/d2l_api/client.py

```diff
@@ -117,6 +117,32 @@ def course_table_of_contents(self, org_unit) -> List[dict]:
         )
         return D2LTableOfContentsSchema(response).parse().get("modules", [])
 
+    def public_url(self, org_unit, file_id) -> str:
+        """
+        Return the URL to download the given file.
+
+        As opposed to other LMS that return a one time signed URL that we can then pass along to via
+        D2L requires us to send the API authentication header to get the file contents.
+
+        To make sure the API authentication header is not expired we'll make a API
+        request here so if it needs to be refreshed we follow the standard token refresh procedure.
+
+        https://docs.valence.desire2learn.com/res/content.html#get--d2l-api-le-(version)-(orgUnitId)-content-topics-(topicId)
+        https://docs.valence.desire2learn.com/res/content.html#get--d2l-api-le-(version)-(orgUnitId)-content-topics-(topicId)-file
```

Comment:
I think this is probably better in code comments next to the code that makes these requests. Otherwise it seems likely that if the code changes this docstring will get out of date.